core/dbt/jsonschemas.py

```diff
@@ -61,13 +61,25 @@ def jsonschema_validate(schema: Dict[str, Any], json: Dict[str, Any], file_path:
     errors: Iterator[ValidationError] = validator.iter_errors(json)  # get all validation errors
 
     for error in errors:
+
         if error.validator == "additionalProperties" and len(error.path) == 0:
             key = re.search(r"'\S+'", error.message)
             deprecations.warn(
                 "custom-top-level-key-deprecation",
                 msg="Unexpected top-level key" + (" " + key.group() if key else ""),
                 file=file_path,
             )
+        elif (
+            error.validator == "anyOf"
+            and (list(error.path))[-1] == "config"
```

Comment:
Not sure if error.path could be empt, but this index operation would throw if it is, and we would not want that.

Reply:
🤔 It could, you are right